overlord/devicestate/devicemgr.go

```diff
@@ -2402,14 +2404,17 @@ func (m *DeviceManager) runFDESetupHook(req *fde.SetupRequest) ([]byte, error) {
 	// that we never run this when the kernel is not fully configured
 	// i.e. when there are no security profiles for the hook
 
-	// state must be locked
+	// state must be unlocked
 	st := m.state
+	st.Lock()
```

Comment:
I'm confused, there's quite a few paths that might end up calling RunFDESetupHook from the boot package as we do boot.RunFDESetupHook = m.runFDESetupHook, maybe we don't have good tests for them but I think more places need to release the lock for this to work but then we hit the issue I mentioned that code in boot is not reentrant/atomic so we need to add some locking there

Reply:
Related to this, one test strategy that might help is when mocking in tests a function that depends directly or indirectly on the lock state is to poke the lock from the mocking impl to check the lock is in the right state